lib/src/functors.ml

```diff
@@ -309,6 +309,7 @@ module Make_native (B :
   type character = char
   type t = B.t
 
+  let max_string_length = Some Sys.max_string_length
```

Comment:
So this imposes that `Make_native` can be used only with `string` or `bytes`?

(I'd imagine easily a bigarray version, like the in the tests)